server/src/main/kotlin/eu/kanade/tachiyomi/network/interceptor/CloudflareInterceptor.kt

```diff
@@ -56,11 +57,38 @@ class CloudflareInterceptor(
             originalResponse.close()
             // network.cookieStore.remove(originalRequest.url.toUri())
 
-            val request =
+            val flareResponseFallback = serverConfig.flareSolverrAsResponseFallback.value
+            val flareResponse =
                 runBlocking {
-                    CFClearance.resolveWithFlareSolverr(setUserAgent, originalRequest)
+                    CFClearance.resolveWithFlareSolver(originalRequest, !flareResponseFallback)
                 }
 
+            if (flareResponse.message.contains("not detected", ignoreCase = true)) {
+                logger.debug { "FlareSolverr failed to detect Cloudflare challenge" }
+
+                if (flareResponseFallback &&
+                    flareResponse.solution.status in 200..299 &&
+                    flareResponse.solution.response != null
+                ) {
+                    val isImage = flareResponse.solution.response.contains(CHROME_IMAGE_TEMPLATE_REGEX)
+                    if (!isImage) {
+                        logger.debug { "Falling back to FlareSolverr response" }
+
+                        setUserAgent(flareResponse.solution.userAgent)
+
+                        return originalResponse
+                            .newBuilder()
+                            .code(flareResponse.solution.status)
+                            .body(flareResponse.solution.response.toResponseBody())
+                            .build()
+                    } else {
+                        logger.debug { "FlareSolverr response is an image html template, not falling back" }
+                    }
+                }
+            }
+
+            val request = CFClearance.requestWithFlareSolverr(flareResponse, setUserAgent, originalRequest)
```

Comment:
I don't really like duplicating requests to FlareSolverr

Reply:
It's not sending duplicate requests, the method name might be causing confusion here. This method has been changed to only craft the new request. It re-uses the flaresolver response from the `resolveWithFlareSolver` call.

Reply:
I see, looks good then